src/cmd/initialize.go

```diff
@@ -137,6 +152,30 @@ func downloadInitPackage(initPackageName, downloadCacheTarget string) error {
 	return nil
 }
 
+// verifyArchitecture verifies that the init package architecture matches the target system architecture.
+func verifyArchitecture(initPackageArch string) {
+	components := pkgConfig.DeployOpts.Components
+	var systemArch string
+	var err error
+
+	// If we're not running in appliance mode (deploying k3s), query the existing cluster for the architecture.
+	// If we are running in appliance mode, get the architecture of the machine we're running on.
+	if !strings.Contains(components, "k3s") {
+		c := cluster.NewClusterOrDie()
+		systemArch, err = c.Kube.GetArchitecture()
+
+		if err != nil {
+			message.Fatal(err, err.Error())
+		}
+	} else {
+		systemArch = runtime.GOARCH
+	}
```

Comment:
Technically for this, the component could be selected interactively without having been specified on the CLI (components would only have k3s in it if it was from the CLI or config file).  A better way to handle this could be comparing the package arch (p.arch) to the cluster arch when the NewClusterWithWait is called

Reply:
The problem I was running into with not checking for k3s/appliance mode was that it was failing saying `Failed to connect to cluster` because `verifyArchitecture` was attempting to set up a k8s client, and if there was no existing cluster to connect to yet, then it fails : https://github.com/defenseunicorns/zarf/actions/runs/4556908999/jobs/8038155551#step:6:3228

So I added in this check to handle the cases where the user is using `zarf init` to deploy a k3s cluster. This current logic fails if the init package arch doesn't match the system arch before zarf actually ever prompts the user after running `zarf init`. If you execute `go run main.go init` on this branch with a mismatched arch package you'll see that it fails before prompting the user to select components. Is there a better way we can handle this part?

Updated this to use the `arch` from the Packager struct. Had to change it to `Arch` so that it would be exported and usable outside of the `packager` package. This required updating any references to this field as well, so there are more files changed now.

Reply:
I'm also noticing that using the arch field from the Packager struct to get the init package architecture isn't working. It's an empty string value, which is causing every case to fail. I'm not exactly sure why it's not populated

Reply:
At this point in the package lifecycle the package hasn't been loaded from disk yet - that happens in loadZarfPkg on L41 of deploy.go https://github.com/defenseunicorns/zarf/blob/main/src/pkg/packager/deploy.go#L41 goes to https://github.com/defenseunicorns/zarf/blob/9846cf4536cb657467be8b8b0a2ccbfdf0d5c7d0/src/pkg/packager/common.go#L192 which goes to https://github.com/defenseunicorns/zarf/blob/9846cf4536cb657467be8b8b0a2ccbfdf0d5c7d0/src/pkg/packager/yaml.go#L21 which sets p.arch

Reply:
Ahh I see. Thanks for pointing that out. Made changes to be able to use p.arch